web/app.js

```diff
@@ -2759,6 +2780,18 @@ function onKeyUp(evt) {
 function onKeyDown(evt) {
   this._isCtrlKeyDown = evt.key === "Control";
 
+  if (
+    this.editorUndoBar?.isOpen &&
+    evt.keyCode !== 9 &&
+    evt.keyCode !== 16 &&
+    (getActiveOrFocusedElement() !== this.editorUndoBar.undoButton ||
+      !(evt.keyCode === 13 || evt.keyCode === 32))
```

Comment:
Also, these two lines should probably be swapped to do the "cheaper" `keyCode` checks *first* to avoid having to unconditionally call `getActiveOrFocusedElement` here.